### PR TITLE
Add detail function app

### PR DESCRIPTION
**Pull request by kaderhashmi**

### CODE REVIEW

File: DFC.Api.JobProfiles.Repository.CosmosDb/CosmosRepository.cs [hunk 1]
```diff
@@ -70,6 +71,31 @@ public async Task<IEnumerable<SummaryDataModel>> GetSummaryListAsync()
...
+        public async Task<IList<SegmentDetailModel>> GetSegmentData(string profileName)
+        {
+            var feedOptions = new FeedOptions
+            {
+                EnableCrossPartitionQuery = true,
+                MaxDegreeOfParallelism = -1,
+            };
+
+            var query = new SqlQuerySpec(
+                "select c.Segments from c where c.CanonicalName =  @profileName",
```
Comment (Muthuramana) [inline DFC.Api.JobProfiles.Repository.CosmosDb/CosmosRepository.cs]: Why are we using SQL query? Why are we not using the same code as segment apps for accessing cosmos database?

File: DFC.Api.JobProfiles/Extensions/StringExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿using Microsoft.AspNetCore.Http;
+
+namespace DFC.Api.JobProfiles.Extensions
+{
+    public static class StringExtensions
+    {
+        public static string PrefixWithBaseAddress(this string url, HttpRequest request)
```
Comment (Muthuramana) [inline DFC.Api.JobProfiles/Extensions/StringExtensions.cs]: Might read better swapping this around.
```suggestion
        public static string GetAbsoluteUrlForRelativePath(this HttpRequest request, string relativePath)
```

**Review by Muthuramana — CHANGES_REQUESTED** [review]

File: DFC.Api.JobProfiles/Extensions/StringExtensions.cs (hunk 2 above)
Comment (kaderhashmi) [reply DFC.Api.JobProfiles/Extensions/StringExtensions.cs]: Swapped as suggested

File: DFC.Api.JobProfiles.Repository.CosmosDb/CosmosRepository.cs (hunk 1 above)
Comment (kaderhashmi) [reply DFC.Api.JobProfiles.Repository.CosmosDb/CosmosRepository.cs]: Made repository generic following discussion

File: DFC.Api.JobProfiles.Data/DataModels/SegmentDataModel.cs [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+using System.Collections.Generic;
+
+namespace DFC.Api.JobProfiles.Data.DataModels
+{
+    public class SegmentDataModel
+    {
+        public List<SegmentDetailModel> Segments { get; set; }
+    }
+
+    public class SegmentDetailModel
```
[the remark this answers is not in the document]
Comment (kaderhashmi) [reply DFC.Api.JobProfiles.Data/DataModels/SegmentDataModel.cs]: Moved

**Review by Muthuramana — APPROVED** [review]

**Pull request merged**